### PR TITLE
Implement resolve user method

### PR DESCRIPTION
**Pull request by isururanawaka**

### CODE REVIEW

**Comment by isururanawaka** [comment]
@machristie here is the PR for resolve user method

File: data-catalog-api/server/src/main/java/org/apache/airavata/datacatalog/api/sharing/SharingManagerImpl.java [hunk 1]
```diff
@@ -89,8 +112,29 @@ public void initialize(String tenantId) throws SharingException {
     }
 
     @Override
-    public UserEntity resolveUser(UserInfo userInfo) {
-        return null;
+    public UserEntity resolveUser(UserInfo userInfo) throws SharingException {
+        try (UserManagementClient userManagementClient = custosClientProvider.getUserManagementClient()) {
```
Comment (machristie) [inline data-catalog-api/server/src/main/java/org/apache/airavata/datacatalog/api/sharing/SharingManagerImpl.java:116]: This is good but it should first check the database (UserRepository) to see if the user already exists or not. Then it only needs to make the Custos API call if it doesn't exist locally.

**Review by machristie — CHANGES_REQUESTED** [review]
Thanks @isururanawaka , this is really helpful. I have one request, see review comment.

I also have a question, is Custos local development mode ready for me to try running Custos locally integrated with Data Catalog? If so can you point me to the README, etc.?

Thanks,

Marcus

File: data-catalog-api/server/src/main/java/org/apache/airavata/datacatalog/api/sharing/SharingManagerImpl.java (hunk 1 above)
Comment (isururanawaka) [reply data-catalog-api/server/src/main/java/org/apache/airavata/datacatalog/api/sharing/SharingManagerImpl.java:116]: @machristie I will add that
Please try this 
https://github.com/apache/airavata-custos/tree/develop

**Comment by machristie** [comment]
Thanks @isururanawaka, I'll take it from here.

**Pull request merged**